components/builder-worker/src/server.rs

```diff
@@ -75,6 +77,12 @@ impl Server {
                 self.fe_sock.connect(&queue)?;
             }
         }
+
+        {
+            let cfg = self.config.read().unwrap();
+            Broker::run(Self::net_ident(), cfg.route_addrs())
+        };
+
```

Comment:
I might be mistaken, but don't you need to `join` on the broker after telling it to run?  All the other servers do that, e.g.

https://github.com/habitat-sh/habitat/blob/master/components/builder-jobsrv/src/server/mod.rs#L228
https://github.com/habitat-sh/habitat/blob/master/components/builder-api/src/server.rs#L56
https://github.com/habitat-sh/habitat/blob/master/components/builder-depot/src/server.rs#L2215
https://github.com/habitat-sh/habitat/blob/master/components/builder-scheduler/src/server/mod.rs#L213
https://github.com/habitat-sh/habitat/blob/master/components/builder-admin/src/server.rs#L55

Curiously enough, sessionsrv doesn't `join` like the rest.  I read the documentation for `std::thread::JoinHandle` which is what `Broker::run()` returns, and I gather that `join` just waits for the thread to finish.  What are the consequences of not calling `join` on the `Broker`?

Reply:
Since we go into an infinite loop right after this, we don't (actually, cannot or even want to) do a join on the broker.  It's probably not the most optimal design, but I didn't want to start refactoring this part of the code.

Reply:
That explains the difficulty I had when I tried to add the `Broker` to this code during the docker integration work.  Thanks for the explanation 👍